### PR TITLE
Split UA synthesis and override into separate exported functions.

### LINKED ISSUE #1: Remove side effects
Attempting to run this code results in an error if an `#output` element doesn't exist in the DOM. Example usage of this retrofill is documented in the README, so would it be possible to remove this chunk of code so this retrofill is usable as a standalone library, not just a demo? 

https://github.com/GoogleChromeLabs/uach-retrofill/blob/fe6b9c60eb3d6ed1b8949d3f52c0c46ccf063223/uach-retrofill.js#L145-L153

### LINKED ISSUE #2: Provide standalone synthesis function
We would like to run an experiment using this retrofill, but would like more control over how `navigator.userAgent`  is updated. More specifically, in addition to retrofilling the `userAgent` string, we also want to track how frequently `navigator.userAgent` is accessed, and whether it is accessed before the retrofill's Promise has resolved.

Would it be possible to either replace the existing `OverrideUserAgentUsingClientHints` function entirely or add an additional function that could be used to synthesize a UA string without actually modifying `navigator.userAgent`? Something like this would be perfect:

```js
/**
* @param {string[]} hints
* @return {Promise<string|undefined>} A Promise that resolves to a string if 
*     a UA could be synthesized from client hints, otherwise undefined.
*/
export async function getUserAgentUsingClientHints(hints) {
  // ...  
}
```

Anyone who wished to simply update `navigator.userAgent` could easily do so using this function:

```js
import {getUserAgentUsingClientHints} from 'uarch-retrofill';

getUserAgentUsingClientHints([""]).then((userAgent) => {
  if (userAgent) { 
      Object.defineProperty(navigator, 'userAgent', {
        value: userAgent,
        writable: false
      });
  }
});
```

### PR DESCRIPTION
**Pull request by rowan-m**
The UA string generation is available as its own function. The override makes use of it. Both functions are exported and a simple `demo.html` is included to try it out.

Fixes #1
Fixes #2

### CODE REVIEW

**Comment by rowan-m** [comment]
No real rewrite of logic here, mostly just replumbing. My wrapping and exporting of promises may be a bit ugly, I can't decide… so, open to better options!

File: uach-retrofill.js [hunk 1]
```diff
@@ -1,42 +1,50 @@
 /**
  * Copyright 2020 Google LLC
  *
- * Licensed under the Apache License, Version 2.0 (the "License");
+ * Licensed under the Apache License, Version 2.0 (the 'License');
```
Comment (miketaylr) [inline uach-retrofill.js:4]: It's cool, I get it. You hate double quotes.

**Review by miketaylr — APPROVED** [review]
Looks good!

I noticed that clicking "architecture" doesn't really do anything (same for model, but that doesn't make sense on Desktop anyways), but I'll file a follow-up -- I think you're just exposing a bug.

File: uach-retrofill.js (hunk 1 above)
Comment (rowan-m) [reply uach-retrofill.js:4]: It"s a compulsion.

**Pull request merged**